argopy/data_fetchers/localftp.py

```diff
@@ -356,4 +358,151 @@ def to_xarray(self, errors: str = 'raise', client=None):
+
+    @property
+    def cachepath(self):
+        """ Return path to cache file for this request """
+        src = self.cachedir
+        file = ("LOCindex_%s.csv") % (self.cname(cache=True))
+        fcache = os.path.join(src, file)
+        return fcache
+
+    def to_dataframe(self):
+        """ filter local index file and return a pandas dataframe """      
+        #  
+        # Try to load cached file if requested:
+        if self.cache and os.path.exists(self.cachepath):
+            df = pd.read_csv(self.cachepath)            
+            return df
+        # No cache found or requested, so we compute:        
+        self.filter_index()
+        #
+        df=pd.read_csv(self.filtered_index)
+        #create datetime & wmo field
+        #local ftp date format 20160513065300
+        df['date'] = pd.to_datetime(df['date'], format="%Y%m%d%H%M%S")
+        df['date_update'] = pd.to_datetime(df['date_update'], format="%Y%m%d%H%M%S")
+
+        df['wmo'] = df.file.apply(lambda x: int(x.split('/')[1]))
+        #
+        # institution & profiler mapping
+        institution_dictionnary=load_dict('institutions')
+        df['tmp1'] = df.institution.apply(lambda x: mapp_dict(institution_dictionnary,x))
+        profiler_dictionnary=load_dict('profilers')
+        df['tmp2'] = df.profiler_type.apply(lambda x: mapp_dict(profiler_dictionnary,x))        
+
+        df=df.drop(columns=['institution','profiler_type'])
+        df=df.rename(columns={"tmp1":"institution","tmp2":"profiler_type"})        
+        
+        # Possibly save in cache for later re-use
+        if self.cache:
+            df.to_csv(self.cachepath,index=False)
+
+        return df
+
+class IndexFetcher_wmo(LocalFTPArgoIndexFetcher):
+    """ Manage access to local ftp Argo data for: a list of WMOs
+
+    """
+    def init(self, WMO: list = [], **kwargs):
+        """ Create Argo data loader for WMOs
+
+            Parameters
+            ----------
+            WMO : list(int)
+                The list of WMOs to load all Argo data for.            
+        """
+        if isinstance(WMO, int):
+            WMO = [WMO] # Make sure we deal with a list        
+        self.WMO = WMO        
+        
+        return self
+
+    def cname(self, cache=False):
+        """ Return a unique string defining the request """
+        if len(self.WMO) > 1:
+            if cache:
+                listname = ["WMO%i" % i for i in self.WMO]                
+                listname = "_".join(listname)                
+            else:
+                listname = ["WMO%i" % i for i in self.WMO]                
+                listname = ";".join(listname)                
+        else:
+            listname = "WMO%i" % self.WMO[0]            
+        return listname
+
+    def filter_index(self):
+        #input file reader
+        inputFileName=self.path_ftp+'/'+self.index_file
+        outputFileName=self.cachedir+'/'+'tmp_'+self.cname()+'.csv'
```

Comment:
```suggestion
        outputFileName = os.path.join(self.cachedir, 'tmp_'+self.cname(cache=True)+'.csv')
```
Also fix this bug in calling cname without cache=True that wotherwise creates files like ``tmp_WMO6902766;WMO6902772.csv``